torch/csrc/distributed/c10d/Types.hpp

```diff
@@ -71,6 +71,14 @@ struct ReduceOp {
     return other == op_;
   }
 
+  bool operator==(const ReduceOp::Kind other) {
```

Comment:
nit: not sure if `Kind` conveys the clearest meaning when a reader reads it somewhere else than where it is defined.
Would `RedOpType` make it clearer? I thought about `OpType` to make it shorter, but it seems `OpType` has been taken in `ProcessGroupNCCL`.